pkg/network/payload/notary_request.go

```diff
@@ -136,3 +136,20 @@ func (r *P2PNotaryRequest) isValid() error {
 	}
 	return nil
 }
+
+// Copy creates a deep copy of the P2PNotaryRequest.
+func (r *P2PNotaryRequest) Copy() *P2PNotaryRequest {
+	var mainTxCopy, fallbackTxCopy *transaction.Transaction
+	mainTxCopy = r.MainTransaction.Copy()
+
+	fallbackTxCopy = r.FallbackTransaction.Copy()
+
+	witnessCopy := r.Witness.Copy()
+
+	return &P2PNotaryRequest{
+		MainTransaction:     mainTxCopy,
+		FallbackTransaction: fallbackTxCopy,
+		Witness:             *witnessCopy,
+		hash:                r.hash,
```

Comment:
Hash initialization is not needed then? To keep this helper consistent with transaction's `Copy()`.